src/main/java/uk/gov/ons/ctp/response/action/service/impl/decorator/PartyActionRequestDecorator.java

```diff
@@ -0,0 +1,157 @@
+package uk.gov.ons.ctp.response.action.service.impl.decorator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.ons.ctp.response.action.message.instruction.ActionContact;
+import uk.gov.ons.ctp.response.action.message.instruction.ActionRequest;
+import uk.gov.ons.ctp.response.action.service.ActionProcessingService;
+import uk.gov.ons.ctp.response.action.service.ActionRequestDecorator;
+import uk.gov.ons.ctp.response.party.representation.Association;
+import uk.gov.ons.ctp.response.party.representation.Attributes;
+import uk.gov.ons.ctp.response.party.representation.Enrolment;
+import uk.gov.ons.ctp.response.party.representation.PartyDTO;
+
+@Slf4j
+public class PartyActionRequestDecorator implements ActionRequestDecorator {
+
+  @Override
+  public void decorateActionRequest(
+      ActionRequest actionRequest, ActionRequestDecoratorContext context) {
+    final PartyDTO parentParty = context.getParentParty();
+    final List<PartyDTO> childParties = context.getChildParties();
+    final Attributes businessUnitAttributes = parentParty.getAttributes();
+    actionRequest.setRegion(businessUnitAttributes.getRegion());
+
+    final ActionContact actionContact = new ActionContact();
+    actionContact.setRuName(businessUnitAttributes.getName());
+    actionContact.setTradingStyle(generateTradingStyle(businessUnitAttributes));
+
+    if (context.getSampleUnitType().isParent() == false) {
+      // BI case - this means the action request is being raised for the child party so
+      // their details are the only ones we're interested in (and we can assume the first and
+      // only element in the childParties list contains the details we are after)
+      PartyDTO childParty = childParties.get(0);
```

Comment:
Maybe two named methods would make this code more self-explanatory and make the comments redundant e.g. `setNonInvididualContactDetails` and `setInvididualContactDetails`.

Reply:
Done